### PR TITLE
Finish onboarding ML inference search request / response processors

### PR DESCRIPTION
**Pull request by ohltyler**
### Description

This PR finishes up what was started in #180. Specifically, it finishes the logic for dynamically generating the ReactFlow workspace nodes/edges, and the workflow template with a `create_search_pipeline` workflow step.
- refactors and adds logic in `workflow_to_template_utils` to generate the `create_search_pipeline` step with any configured search request and search response processors. Also updates the `create_index` step to gracefully handle the cases of no ingest or search pipeline defined, which is allowed in the UI.
- refactors and adds logic in the base `utils` to generate the dynamic list of nodes/edges in the search flow

Demo video showing the workspace flow (template configuration and testing different edge cases of search and ingest pipelines was verified manually - can't test as there is no final implementation of the ML processors on the search side, yet):

[screen-capture (45).webm](https://github.com/opensearch-project/dashboards-flow-framework/assets/62119629/4840b73b-ccb6-4c67-aebf-48f5cae3d512)

### Issues Resolved
Makes progress on #23 

### Check List
- [x] Commits are signed per the DCO using `--signoff`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/pages/workflow_detail/utils/workflow_to_template_utils.ts [hunk 1]
```diff
@@ -59,12 +70,66 @@ function configToProvisionTemplateFlow(config: WorkflowConfig): TemplateFlow {
...
+          user_inputs: {
+            pipeline_id: searchPipelineName,
+            configurations: {
+              request_processors: searchRequestProcessors,
+              response_processors: searchResponseProcessors,
+            },
+          },
+        } as CreateSearchPipelineNode,
+      ]
+    : [];
```
Comment (owaiskazi19) [inline public/pages/workflow_detail/utils/workflow_to_template_utils.ts:124]:
```suggestion
  const hasProcessors = searchRequestProcessors.length > 0 || searchResponseProcessors.length > 0;

return hasProcessors
  ? [
      {
        id: searchPipelineName,
        type: WORKFLOW_STEP_TYPE.CREATE_SEARCH_PIPELINE_STEP_TYPE,
        user_inputs: {
          pipeline_id: searchPipelineName,
          configurations: {
            request_processors: searchRequestProcessors,
            response_processors: searchResponseProcessors,
          },
        },
      } as CreateSearchPipelineNode,
    ]
  : [];
```

File: public/pages/workflow_detail/utils/workflow_to_template_utils.ts [hunk 2]
```diff
@@ -100,49 +165,54 @@ function processorConfigsToTemplateNodes(
...
+  if (searchPipelineNode) {
+    finalPreviousNodeInputs = {
+      ...finalPreviousNodeInputs,
+      [searchPipelineNode.id]: 'pipeline_id',
+    };
+    finalSettings = {
+      ...finalSettings,
+      ['index.search.default_pipeline']: `\${{${searchPipelineNode.id}.pipeline_id}}`,
+    };
+  }
```
Comment (owaiskazi19) [inline public/pages/workflow_detail/utils/workflow_to_template_utils.ts:206]: To avoid duplication. Something like:
```suggestion
  const updateFinalInputsAndSettings = (node) => {
  if (node) {
    finalPreviousNodeInputs = {
      ...finalPreviousNodeInputs,
      [node.id]: 'pipeline_id',
    };

    const pipelineKey = node.type === WORKFLOW_STEP_TYPE.CREATE_INGEST_PIPELINE_STEP_TYPE
      ? 'default_pipeline'
      : 'index.search.default_pipeline';

    finalSettings = {
      ...finalSettings,
      [pipelineKey]: `\${{${node.id}.pipeline_id}}`,
    };
  }
};

updateFinalInputsAndSettings(ingestPipelineNode);
updateFinalInputsAndSettings(searchPipelineNode);
```

**Review by owaiskazi19 — COMMENTED** [review]
Minor refactoring comments. Rest LGTM

File: public/pages/workflow_detail/utils/workflow_to_template_utils.ts (hunk 1 above)
Comment (ohltyler) [reply public/pages/workflow_detail/utils/workflow_to_template_utils.ts:124]: Sure, will update. Typically I prefer to minimize the number of verbose variables hence doing it inline, will add this one explicitly if it helps for readability.

File: public/pages/workflow_detail/utils/workflow_to_template_utils.ts (hunk 2 above)
Comment (ohltyler) [reply public/pages/workflow_detail/utils/workflow_to_template_utils.ts]: Sure, thanks for the suggestion! will update.

File: public/pages/workflow_detail/utils/workflow_to_template_utils.ts (hunk 1 above)
Comment (ohltyler) [reply public/pages/workflow_detail/utils/workflow_to_template_utils.ts]: updated in latest commit

**Review by owaiskazi19 — APPROVED** [review]
Thanks for addressing the comments
